src/test/java/de/atextor/turtle/formatter/TurtleFormatterTest.java

```diff
@@ -719,6 +737,301 @@ void testRdfTypeAsObjectIsValid() {
+        final Model model = modelFromString( modelString );
+
+        final FormattingStyle style = FormattingStyle.DEFAULT;
+        final TurtleFormatter formatter = new TurtleFormatter( style );
+        final String result = formatter.apply( model );
+        assertThat( result.trim() ).isEqualTo( modelString.trim() );
+    }
+
+    @Test
+    void testRdfListAnonymous(){
+        final String modelString = """
+            @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+            @prefix ex: <http://example.com/ns#> .
+            
+            ex:something a ex:Thing ;
+              ex:hasList ( ex:one ex:two ) .
+            """;
+        final Model model = modelFromString( modelString );
+
+        final FormattingStyle style = FormattingStyle.DEFAULT;
+        final TurtleFormatter formatter = new TurtleFormatter( style );
+        final String result = formatter.apply( model );
+        assertThat( result.trim() ).isEqualTo( modelString.trim() );
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void testConsistentBlankNodeOrdering(String content){
+        final FormattingStyle style = FormattingStyle.DEFAULT;
+        final TurtleFormatter formatter = new TurtleFormatter( style );
+        for (int i = 0; i < 1; i++) {
+            final String result = formatter.applyToContent(content);
+            assertThat(result.trim()).isEqualTo(content.trim());
+        }
+    }
+
+    static Stream<Arguments> testConsistentBlankNodeOrdering(){
+        return Stream.of(
+    """
+            @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+            @prefix ex: <http://example.com/ns#> .
+            
+            [ 
+              a ex:Something ;
+            ] .
+            
+            [
+              a ex:SomethingElse ;
+            ] .
+            """,
+
+            """
+            @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+            @prefix ex: <http://example.com/ns#> .
+            
+            ex:aThing ex:has [
+                a ex:Something ;
+              ] ;
+              ex:has [
+                a ex:SomethingElse ;
+              ] .
+            """
+            ,
+           """
+           @prefix ex: <http://example.com/ns#> .
+           
+           _:blank1 ex:has [
+               ex:has _:blank1 ;
+             ] ."""
+        ).map(s -> Arguments.of(s));
+    }
+
+    @Test
+    void testPreviouslyIdentifiedBlankNode(){
+        String content =   """
+           @prefix ex: <http://example.com/ns#> .
+           
+           _:gen0 ex:has [
+               ex:has _:gen0 ;
+             ].""";
+        String expected = """
+          @prefix ex: <http://example.com/ns#> .
+           
+          _:gen0 ex:has [ 
+              ex:has _:gen0 ;
+            ] .""";
+        final FormattingStyle style = FormattingStyle.DEFAULT;
+        final TurtleFormatter formatter = new TurtleFormatter( style );
+        for (int i = 0; i < 20; i++) {
```

Comment:
I see what you're doing here but I wonder if it's necessary _or_ sufficient. It might be a good idea to turn this logic into proper property-based tests in the [jqwik](https://jqwik.net/)-based TurtleFormatterPropertyTest.java, but you can consider this out of scope for this PR.

Reply:
I'll leave it as-is then (for this PR anyway). These loops were necessary (and sufficient) while implementing the constant blank node serialization code: it seems the probability for different ordering over the runs are uniformly distributed.